### DISCUSSION TITLE
OTEL format - how can it handle enums

### DISCUSSION THREAD

**Discussion by snimmagadda-f5** [open-telemetry/community#799] [Community Discussions]
I have a requirement where there is existing data containing enums that need to be converted to OTEL

data like below - metric name size, which can be mapped to form 1 to 3
HIGH -1
MEDIUM - 2
LOW -3

Even if I use IntDataPoint.Labels for string and IntDataPoint.Value for value, how can I inform the exporter about such a thing

**Comment by Oberon00** [comment]
> how can I inform the exporter about such a thing

What is you goal here? Why would you want to inform the exporter? Does just sending the data as either an integer or string not work for you?

**Reply by snimmagadda-f5** [reply]
sending as string will work for me, but I don't see any support for non-numeric values in OTEL format

**Reply by Oberon00** [reply]
It is not entirely clear to me what you mean by "OTEL format". The attribute types supported by the OpenTelemetry APIs are described here: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/common.md#attributes Strings are supported. Maybe you have a problem with a particular language implementation?

**Reply by snimmagadda-f5** [reply]
I am trying to build a receiver, that would have received data in a custom format (JSON), containing numeric values and strings.
So building "pdata.Metrics" structure

Can I just use - IntDataPoint.Labels for string and would that be fine? without setting -  IntDataPoint.Value ?
would that work?

**Comment by tigrannajaryan** [comment]
@snimmagadda-f5 if I understand correctly you want to be able to record the value of the data point such that it is from a set of known values (enum). If so then there is no such capability currently. Otel currently supports integer or double gauges, but there is no way to tell that the integer value belongs to a constrained set of allowed values.
I think using labels is not the right approach, labels are for recording dimensions of metrics. If you put the enum's name into labels it would result in different timeseries per each distinct value recorded, essentially you will get N timeseries, each timeseries having the exact same value recorded at some different points of time.

@open-telemetry/specs-metrics-approvers FYI, something to consider as possible data type for metrics.

**Reply by Oberon00** [reply]
OK, I did not realize in the beginning that this was about metric values. Good point, I think the keyword would be "categorical variable". https://en.wikipedia.org/wiki/Categorical_variable
